### PR TITLE
bpf: add batch table get

### PR DESCRIPTION
**Pull request by rootfs**
this is ported from #665

### CODE REVIEW

File: pkg/collector/container_hc_collector.go [hunk 1]
```diff
@@ -109,23 +116,52 @@ func (c *Collector) updateBPFMetrics() {
...
 
-		// if ebpf map batch deletion operation is supported we add the key to the list otherwise we delete the key
-		if config.EnabledBPFBatchDelete {
-			keysToDelete = append(keysToDelete, key)
-		} else {
-			err = c.bpfHCMeter.Table.Delete(key) // deleting the element to reset the counter values
-			if err != nil {
-				klog.Infof("could not delete bpf table elemets, err: %v", err)
+		// if not deleted during get, delete it now
+		if !ebpfBatchGetAndDelete {
```
Comment (jichenjc) [inline pkg/collector/container_hc_collector.go:156]: if this is false, when do we do the delete operation? if will not append to the delete list or delete directly ...

Comment (marceloamaral) [reply pkg/collector/container_hc_collector.go:156]: @rootfs should this line be `if ebpfBatchGetAndDelete {`

The line 203 is the `!ebpfBatchGetAndDelete` case, right?

**Review by marceloamaral — COMMENTED** [review]
Looks good to me. Only have one comment.

File: pkg/collector/container_hc_collector.go (hunk 1 above)
Comment (rootfs) [reply pkg/collector/container_hc_collector.go:156]: this line is right, if `ebpfBatchGetAndDelete == true`, all the keys are already deleted in the batch get

**Comment by rootfs** [comment]
@jichenjc @marceloamaral updated, PTAL thanks

**Comment by jichenjc** [comment]
/lgtm

I assume the original result posted in the origianl issue still valid after the above fix?

**Pull request merged**